internal/builder/main.go

```diff
@@ -50,6 +50,10 @@ func GenerateAndCompile(cfg Config) error {
 
 // Generate assembles a new distribution based on the given configuration
 func Generate(cfg Config) error {
+	// create a warning message for non-aligned builder and collector base
+	if cfg.Distribution.OtelColVersion != defaultOtelColVersion {
+		cfg.Logger.Info("You're building a distribution with non-aligned version of the builder. Compilation may fail due to API changes. Please upgrade your builder or API")
```

Comment:
How about mentioning the versions in the message? Like, the builder's otelcol version and the distribution otelcol version.